beater/server.go

```diff
@@ -101,13 +101,13 @@ func createHandler(p processor.Processor, config Config, publish successCallback
 
 		// Limit size of request to prevent for example zip bombs
 		limitedReader := io.LimitReader(reader, config.MaxUnzippedSize)
-		err = p.Validate(limitedReader)
+		payload, err := p.Validate(limitedReader)
 		if err != nil {
 			sendError(w, r, 400, fmt.Sprintf("Data Validation error: %s", err), true)
 			return
 		}
 
-		list := p.Transform()
+		list := payload.Transform()
```

Comment:
I considered that, but it means the interface will have a method: `Transform(Payload)` which takes a `Payload` interface. If we remove the `Transform` method from the `Payload` interface, there's nothing left. So it take an empty interface, which could be anything. 

Also, the `Transform` method would look like this. Notice the need to type assert `pr.Payload` to an actual struct:

```
func (p *processor) Transform(payload pr.Payload) []beat.Event {
	var events []beat.Event
	pa := payload.(Payload)
	for _, tx := range pa.Events {

		events = append(events, pr.CreateDoc(tx.Mappings(pa)))

		for _, tr := range tx.Traces {
			events = append(events, pr.CreateDoc(tr.Mappings(pa, tx)))
		}
	}

	return events
}
```